mitodl/mit-open
# remove privacy level from user lists and learning paths

### What are the relevant tickets?
Closes https://github.com/mitodl/mit-open/issues/1431
Closes https://github.com/mitodl/hq/issues/5208

### Description (What does it do?)
This PR removes the privacy level display from the "Add to User List" dialog, as well as the "privacy level" option from the learning path edit / add dialog. Learning paths will default to the "private" level, making them not visible to unauthenticated users at the learning path URL.

### Screenshots (if appropriate):
![image](https://github.com/user-attachments/assets/08c94e31-046b-4093-82b3-fc84523a96a7)
![image](https://github.com/user-attachments/assets/5bd79e37-db5e-4d84-b614-219f6f315d5b)
![image](https://github.com/user-attachments/assets/509e5e83-7f27-4458-bbbe-bd16bc086e06)
![image](https://github.com/user-attachments/assets/0f402cae-99a1-470b-8e9d-d9e24ba981fd)
![image](https://github.com/user-attachments/assets/e8247ed7-6095-40f6-b708-b747723e526c)
![image](https://github.com/user-attachments/assets/5bec5f53-8e49-4058-a8ff-f5641e9f745f)

### How can this be tested?
 - Spin up `mit-open` on this branch
 - Ensure that you have a user that is marked as a staff or superuser and log in as said user
 - Click upper right icon to open the drawer and click on Learning Paths
 - Click on "Create new list" and test adding a new learning path
 - Click on your learning path after you create it to open the detail page
 - Click on "Edit List," change some of the properties and save
 - Verify that the list saved correctly and everything works as expected
 - Visit the search page at http://localhost:8063/search
 - Click the bookmark icon on any of the search results to bring up the Add to User List dialog
 - If you don't have any User Lists, create one
 - Add a few items to your User List
 - Verify that the privacy level pill is not displayed in the dialog

### Additional Context
There are some additional issues here that are out of scope:

 - If you select enough subjects in the learning path creation dialog, they visually overflow outside the bounds of the dropdown as evidenced in the screenshots
